### PR TITLE
Write checklist for teammates

### PR DESCRIPTION
**Pull request by NathanLovato**
This is a draft for the checklist for the team. @EadhCaillin could you take a look at it and tell me what you think it's missing?

Note there's only one new document, the checklist. Changes in other documents tweak the order in the site's menu, to place the checklist after the guidelines.

### CODE REVIEW

**Comment by NathanLovato** [comment]
What would you recommend? I'd like to go for what will maximize the tutors' efficiency in practice.

What do you think @Xananax ? I'd like to know what works best for the two of you based on your experience.

Context for you Jad: after making all 5-6 updated guideline documents for GDQuest (code style, writing style, how to structure tutorials, etc.) we're working on a checklist for every current and future team member to use to check their work - code and writing.

**Comment by EadhCaillin** [comment]
I vote for a detailed checklist with nested lists where necessary. 
Covering all the points in the guidelines. 
Arranged in the natural order of progress.

**Comment by Xananax** [comment]
Just to be clear, is the discussion's about choosing between:

1. a shortlist with links to other checklists, or
2. a large itemized checklist with potentially indented checklists?

If yes, I vote for 2. I'd even suggest to have the list as a template doc that one can (should) copy in a project. People can then refer to it in the editor, and check things as they go. Plugins that extract todo lists (there are a few for each editor) can even keep this in a sidebar.

Further thoughts:

- Possibly add the checklist to `.gitignore` by default. I keep my own `TODO.md` in .gitignore, for example. But maybe in that case, it's good to have that info committed (so progress is clear)
. Potentially have the list as a printable doc for the few souls that still use printers, or like to hand check things done on their e-ink device

**Comment by NathanLovato** [comment]
Thanks for your feedback, let's go for the detailed checklist then!

@EadhCaillin is this something you could tackle, building upon this PR (or re-writing it if the points in this PR aren't valid)? Or should I flesh it out myself?

**Comment by EadhCaillin** [comment]
What Xananax is saying is what I meant by interactive the other day.

**Comment by EadhCaillin** [comment]
@NathanLovato, yes I can. If you like, you can publish what you have now until I get to it.

**Comment by EadhCaillin** [comment]
@NathanLovato, the complete checklist is ready for review. I made minor edits in some of the guides I used along the way.

File: content/docs/guidelines/best-practices/checklist/index.md [hunk 1]
```diff
@@ -1,28 +1,225 @@
 +++
 title = "The checklist"
 description = ""
-author = "nathan"
+author = "sally"
```
Comment (NathanLovato) [inline content/docs/guidelines/best-practices/checklist/index.md:4]: We'll need to add author data for you, a file like this: https://github.com/GDQuest/website/blob/master/data/authors/johnny.toml

Could you send me that or create such a file in data/authors? It should be named `sally.toml` in this case.

**Comment by NathanLovato** [comment]
It's looking good. Just the long list of "Did you" makes everything longer to read.

How about having everything as a series of actions?

> - [ ] Did you keep your code as simple and readable as possible?
>     - [ ] Did you avoid making your code flexible and future-proof?
>     - [ ] Did you pick one valid solution?
>     - [ ] Did you favor inline code in functions over splitting code across many functions?

->

> - [ ] The code is as simple and readable as possible
>     - [ ] I didn't future-proof
>     - [ ] I picked one valid solution
>     - [ ] I favored inline code in functions over splitting code across many functions

**Comment by EadhCaillin** [comment]
Ok. Will you change it or should I? 
(I did occur to me. I had opted against to stick to a reminder and avoid a certification or contractual tone).

**Comment by NathanLovato** [comment]
I can change it if you'd like, you can if you'd prefer to do it yourself. Just let me know

File: content/docs/guidelines/best-practices/checklist/index.md (hunk 1 above)
Comment (EadhCaillin) [reply content/docs/guidelines/best-practices/checklist/index.md:4]: Ok, you can change the author name to Nathan for now.

**Comment by EadhCaillin** [comment]
I changed the sentence structures. I'll let you take it from there? You can make any further changes and merge or let me know if you'd like me to recheck.

File: content/docs/guidelines/best-practices/checklist/index.md (hunk 1 above)
Comment (EadhCaillin) [reply content/docs/guidelines/best-practices/checklist/index.md:4]: I changed it back :)

**Comment by EadhCaillin** [comment]
Reread and edited with a bit more distance. Really done from my end now. It's yours to review, edit and merge.

**Comment by NathanLovato** [comment]
Thanks much, it's great!

**Pull request merged**